ui/test/nuxeo-image-viewer.test.js

```diff
@@ -0,0 +1,326 @@
+      });
+    });
+  });
+
+  suite('Interactions', () => {
+    suite('Controls Toolbar', () => {
+      let viewer;
+      let toolbar;
+      setup(async () => {
+        viewer = await fixture(
+          html`
+            <nuxeo-image-viewer src="${base}/resources/sample.png" controls></nuxeo-image-viewer>
+          `,
+        );
+        await viewerLoad(viewer);
+        expect(isToolbarVisible(viewer)).to.be.true;
+        toolbar = viewer.$$('#toolbar');
+      });
+
+      test('Should call method to zoom out when first button is used', () => {
+        const zoomOutOption = toolbar.children[0];
+        expect(zoomOutOption.getAttribute('data-action')).to.equal('zoom-out');
+
+        sinon.spy(viewer._el, 'zoom');
+        tap(zoomOutOption);
+        expect(viewer._el.zoom.withArgs(-0.1).calledOnce).to.be.true;
+      });
+
+      test('Should call method to fit to real size when second button is used and no zoom was applied', () => {
+        const fitToRealSizeOption = toolbar.children[1];
+        expect(fitToRealSizeOption.getAttribute('data-action')).to.equal('fit-to-real-size');
+
+        sinon.spy(viewer._el, 'zoomTo');
+        tap(fitToRealSizeOption);
+        expect(viewer._el.zoomTo.withArgs(1).calledOnce).to.be.true;
+      });
+
+      test('Should call method to fit to viewer when second button is used and some zoom was applied', () => {
+        viewer._fitToRealSize = true;
+
+        const fitToViewerOption = toolbar.children[1];
+        expect(fitToViewerOption.getAttribute('data-action')).to.equal('fit-to-viewer');
+
+        sinon.spy(viewer._el, 'zoomTo');
+        tap(fitToViewerOption);
+        expect(viewer._el.zoomTo.calledOnce).to.be.true;
+      });
+
+      test('Should call method to zoom in when third button is used', () => {
+        const zoomInOption = toolbar.children[2];
+        expect(zoomInOption.getAttribute('data-action')).to.equal('zoom-in');
+
+        sinon.spy(viewer._el, 'zoom');
+        tap(zoomInOption);
+        expect(viewer._el.zoom.withArgs(0.1).calledOnce).to.be.true;
+      });
+
+      test('Should call method to rotate left when fourth button is used', () => {
+        const rotateLeftOption = toolbar.children[3];
+        expect(rotateLeftOption.getAttribute('data-action')).to.equal('rotate-left');
+
+        sinon.spy(viewer._el, 'rotate');
+        tap(rotateLeftOption);
+        expect(viewer._el.rotate.withArgs(-90).calledOnce).to.be.true;
+      });
+
+      test('Should call method to rotate right when fifth button is used', () => {
+        const rotateRightOption = toolbar.children[4];
+        expect(rotateRightOption.getAttribute('data-action')).to.equal('rotate-right');
+
+        sinon.spy(viewer._el, 'rotate');
+        tap(rotateRightOption);
+        expect(viewer._el.rotate.withArgs(90).calledOnce).to.be.true;
+      });
+    });
+
+    suite('Window Resize', () => {
+      test("Should take no action when viewer's parent width is zero", async () => {
```

Comment:
I'm wondering if "take no action" might seem to abstract. Should we specify it? Cannot think on an good alternative though 🤔

Reply:
Indeed "take no action" doesn't give any clue to what the test is checking. Perhaps we could use "not resize" (which is the event our element listens too) or "reset zoom" (which is what should not happen), or even both, but probably it would be too lengthy:
```suggestion
      test("Should not reset zoom when viewer's parent width is zero", async () => {
```
(applies to comments below)